File: go/difference-of-squares/difference_of_squares.go
package diffsquares

const testVersion = 1

// SquareOfSums calculates the square of the sum of the first `count` terms
func SquareOfSums(count int) (value int) {
	for term := 1; term <= count; term++ {
		value += term
	}

	value *= value

	return
}

// SumOfSquares calculates the sum of the squares of the first `count` terms
func SumOfSquares(count int) (value int) {
	for term := 1; term <= count; term++ {
		value += term * term
	}

	return
}

// Difference calculates the difference between the square of the sum and the sum of the squares
func Difference(count int) (difference int) {
	return SquareOfSums(count) - SumOfSquares(count)
}

File: ruby/difference-of-squares/difference_of_squares.rb
# Squares defines methods to calculate the Difference Of Squares
class Squares
  def initialize(number)
    @range = 1..number
  end

  # square_of_sum calculates the square of the sum the terms in the series
  def square_of_sum
    @range.sum**2
  end

  # square_of_sum calculates the sum of the squares of the terms in the series
  def sum_of_squares
    @range.map { |n| n**2 }.sum
  end

  # difference calculates the difference between
  # the square of the sum of the terms and the sum of the squares of the terms
  def difference
    square_of_sum - sum_of_squares
  end
end

module BookKeeping
  VERSION = 4
end
